platform/ios/app-swift/Sources/LocationPrivacyExample.swift

```diff
@@ -9,75 +9,114 @@ enum LocationAccuracyState {
+        self.mapViewModel = mapViewModel
+        super.init()
+        self.mapViewModel.mapCoordinater = self
+    }
+
+    @MainActor func mapView(_: MLNMapView, didChangeLocationManagerAuthorization manager: MLNLocationManager) {
         guard let accuracySetting = manager.accuracyAuthorization else {
             return
         }
 
         switch accuracySetting() {
         case .fullAccuracy:
-            locationAccuracy = .fullAccuracy
+            mapViewModel.locationAccuracy = .fullAccuracy
         case .reducedAccuracy:
-            locationAccuracy = .reducedAccuracy
+            mapViewModel.locationAccuracy = .reducedAccuracy
         @unknown default:
-            locationAccuracy = .unknown
+            mapViewModel.locationAccuracy = .unknown
         }
     }
 
     // when a location is available for the first time, we fly to it
     func mapView(_ mapView: MLNMapView, didUpdate _: MLNUserLocation?) {
-        if pannedToUserLocation {
-            return
-        }
+        guard !pannedToUserLocation else { return }
         guard let userLocation = mapView.userLocation else {
             print("User location is currently not available.")
             return
         }
         mapView.fly(to: MLNMapCamera(lookingAtCenter: userLocation.coordinate, altitude: 100_000, pitch: 0, heading: 0))
+        pannedToUserLocation = true
+    }
+
+    func mapView(_ mapView: MLNMapView, didFinishLoading _: MLNStyle) {
+        if self.mapView != mapView {
+            self.mapView = mapView
+        }
+    }
+
+    func mapViewDidFinishLoadingMap(_ mapView: MLNMapView) {
+        if self.mapView != mapView {
+            self.mapView = mapView
+        }
+    }
+
+    deinit {
+        // Ensure the coordinator is deallocated if MapLibreViewRepresentable.dismantleUIView was called
     }
 }
 
-struct LocationPrivacyExample: UIViewRepresentable {
-    @Binding var mapView: MLNMapView
-    @Binding var locationAccuracy: LocationAccuracyState
+struct MapLibreViewRepresentable: UIViewRepresentable {
+    let mapViewModel: MapViewModel
 
-    func makeCoordinator() -> Coordinator {
-        Coordinator(mapView: $mapView, locationAccuracy: $locationAccuracy)
+    func makeCoordinator() -> MapLibreRepresentableCoordinator {
+        MapLibreRepresentableCoordinator(mapViewModel: mapViewModel)
     }
 
     func makeUIView(context: Context) -> MLNMapView {
         let mapView = MLNMapView()
-        mapView.showsUserLocation = true
+        // Set delegate first, otherwise delegate callbacks can be missed. For example, setting showsUserLocation calls mapView:didUpdate:userLocation
         mapView.delegate = context.coordinator
-
+        mapView.showsUserLocation = true
         return mapView
     }
 
-    func updateUIView(_: MLNMapView, context _: Context) {}
+    func updateUIView(_: MLNMapView, context _: Context) {
+        // Be careful with doing any real work in updateUIView, this is called a lot.
```

Comment:
https://developer.apple.com/documentation/swiftui/uiviewrepresentable/updateuiview(_:context:) states that all UI updates are to be made here. we are provided with the view and the context here. By performing updates here no own reference to the mapview has to be held.

Yes it is called, but "a lot" is relative. Its not called 120 times a second, its called when SwiftUI has been informed that something regarding the state of this view has been changed, so its the time to see if one needs to update this view.

I don't think it's wise to discourage an Apple best practice in the MapLibre docs.

Reply:
I agree for this PR and this example that is out of scope. Updated the PR to follow Apple's doc. 

I need to write a blog post about all this stuff I've found out with `UIViewRepresentable` and put my options there.